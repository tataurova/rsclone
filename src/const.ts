export const RouteName = {
  MAIN: '/',
  LOGIN: '/login',
  GONE: '/gone',
  LOOKING_RELATIVES: '/looking-relatives',
  ACTIVE_SEARCHES: '/active-searches',
  CLOSED_SEARCHES: '/closed-searches',
};
export const PageName = {
  MAIN: '/',
  LOGIN: '/login',
  GONE: '/gone',
  LOOKING_RELATIVES: '/looking-relatives',
  ACTIVE_SEARCHES: '/active-searches',
  CLOSED_SEARCHES: '/closed-searches',
};

export const AuthorizationStatus = {
  AUTH: 'AUTH',
  NO_AUTH: 'NO_AUTH',
};

export const Error = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
  NOT_FOUND: 404,
  SERVICE_UNAVAILABLE: 503,
  TIMEOUT_ERROR: 'timeout_error',
};

export const headCells = [
  { id: 'status', label: 'Статус' },
  { id: 'city', label: 'Город' },
  { id: 'name', label: 'ФИО' },
  { id: 'age', label: 'Возраст' },
  { id: 'date', label: 'Дата' },
  { id: 'coordinator', label: 'Координатор' },
  { id: 'people', label: 'Людей на поиске' },
  { id: 'actions', label: 'Действия', disableSorting: true },
];

export const getStatusCollection = [
  'Резерв',
  'Поиск на месте',
  'Стоп. Проверка информации',
  'Архив',
];

export const SearcherStatus = {
  way: 'В пути',
  place: 'На месте',
  home: 'Дома',
};

export const TransportStatus = {
  car: 'На машине',
  own: 'Своим ходом',
  need: 'Нужна машина',
};

export const MAX_TIMEOUT = 5000;
export const SHOW_ERROR_TIMEOUT = 3000;
