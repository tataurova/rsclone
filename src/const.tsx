export const PageName = {
  MAIN: '/',
  LOGIN: '/login',
  GONE: '/gone',
  LOOKING_RELATIVES: '/looking-relatives',
  ACTIVE_SEARCHES: '/active-searches',
  CLOSED_SEARCHES: '/closed-searches',
};
