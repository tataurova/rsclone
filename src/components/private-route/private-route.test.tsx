import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { mount, configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { AuthorizationStatus, RouteName } from '../../const';
import CardPage from '../card-page/card-page';
import PrivateRoute from './private-route';

const mockStore = configureStore([]);
configure({ adapter: new Adapter() });

describe('<PrivateRoute />', () => {
  it('Should PrivateRoute render correctly', () => {
    const initialState = {
      AUTH: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    };
    const store = mockStore(initialState);
    const tree = renderer.create(
            <BrowserRouter>
                <PrivateRoute
                    store={store}
                    path={RouteName.GONE}
                    exact={true}
                    render={() => (
                            <CardPage
                                page={'1'}
                            />
                    )}
                />
            </BrowserRouter>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Renders the component when user with auth', () => {
    const initialState = {
      AUTH: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    };
    const store = mockStore(initialState);

    const privateRoute = mount(
            <BrowserRouter>
                <PrivateRoute
                    store={store}
                    path={RouteName.GONE}
                    exact={true}
                    render={() => (
                            <CardPage
                                page={'1'}
                            />
                    )}
                />
            </BrowserRouter>,
    );
    expect(privateRoute.length).toEqual(1);
  });
});
