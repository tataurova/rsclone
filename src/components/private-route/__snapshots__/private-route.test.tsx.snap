// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> Should PrivateRoute render correctly 1`] = `null`;
